contrib/src/main/java/com/datatorrent/contrib/parser/JsonParser.java

```diff
@@ -0,0 +1,187 @@
+ * 
+ * 
+ * @displayName JsonParser
+ * @category Parsers
+ * @tags json pojo parser
+ * @since 3.2.0
+ */
+@InterfaceStability.Evolving
+public class JsonParser extends Parser<String, KeyValPair<String, String>>
+{
+
+  /**
+   * Contents of the schema.Schema is specified as per http://json-schema.org/
+   */
+  private String jsonSchema;
+  private transient JsonSchema schema;
+  private transient ObjectMapper objMapper;
+  /**
+   * output port to emit validate records as JSONObject
+   */
+  public transient DefaultOutputPort<JSONObject> parsedOutput = new DefaultOutputPort<JSONObject>();
+  /**
+   * metric to keep count of number of tuples emitted on {@link #parsedOutput}
+   * port
+   */
+  @AutoMetric
+  long parsedOutputCount;
+
+  @Override
+  public void beginWindow(long windowId)
+  {
+    super.beginWindow(windowId);
+    parsedOutputCount = 0;
+  }
+
+  @Override
+  public void setup(OperatorContext context)
+  {
+    try {
+      if (jsonSchema != null) {
+        JsonSchemaFactory factory = JsonSchemaFactory.byDefault();
+        JsonNode schemaNode = JsonLoader.fromString(jsonSchema);
+        schema = factory.getJsonSchema(schemaNode);
+      }
+      objMapper = new ObjectMapper();
+    } catch (ProcessingException | IOException e) {
+      DTThrowable.wrapIfChecked(e);
+    }
+  }
+
+  @Override
+  public void processTuple(String tuple)
+  {
+    try {
+      if (schema != null) {
+        ProcessingReport report = null;
+        JsonNode data = JsonLoader.fromString(tuple);
+        report = schema.validate(data);
+        if (report != null && !report.isSuccess()) {
+          Iterator<ProcessingMessage> iter = report.iterator();
+          StringBuilder s = new StringBuilder();
+          while (iter.hasNext()) {
+            ProcessingMessage pm = iter.next();
+            s.append(pm.asJson().get("instance").findValue("pointer")).append(":").append(pm.asJson().get("message"))
+                .append(",");
+          }
+          s.setLength(s.length() - 1);
+          errorTupleCount++;
+          err.emit(new KeyValPair<String, String>(tuple, s.toString()));
+          return;
+        }
+      }
+      if (parsedOutput.isConnected()) {
+        parsedOutput.emit(new JSONObject(tuple));
+        parsedOutputCount++;
+      }
+      if (out.isConnected()) {
+        out.emit(objMapper.readValue(tuple, clazz));
+        emittedObjectCount++;
+      }
+    } catch (JSONException | ProcessingException | IOException e) {
+      errorTupleCount++;
+      err.emit(new KeyValPair<String, String>(tuple, e.getMessage()));
+      logger.error("Failed to parse json tuple {}, Exception = {} ", tuple, e);
+    }
+  }
+
+  @Override
+  public Object convert(String tuple)
+  {
+    throw new UnsupportedOperationException("Not supported");
+  }
+
+  @Override
+  public KeyValPair<String, String> processErrorTuple(String input)
+  {
+    throw new UnsupportedOperationException("Not supported");
```

Comment:
Again, can you please return null here and log it, instead of throwing exception. This call (if I understand) correctly is coming from super class. If tomorrow super class changes, this should not get impacted.